pkg/state/state_skel.go

```diff
@@ -266,12 +266,48 @@ func (s *stateSkel) handleStateObjectsDeletion(ctx context.Context) (SyncState,
 	return SyncStateIgnore, nil
 }
 
-func (s *stateSkel) deleteStateRelatedObjects(ctx context.Context) (bool, error) {
+// remove stale object of the state, returns boolean which indicates if removal is in progress and
+// an error if failed to remove an object
+func (s *stateSkel) handleStaleStateObjects(ctx context.Context,
+	currentObjs []*unstructured.Unstructured) (bool, error) {
```

Comment:
nit: currentObjs -> desiredObjs ?

currentObjs sounds like the current objs we got in the cluster.

Reply:
done